nptdms/tdms_segment.py

```diff
@@ -488,28 +520,30 @@ def _read_data_chunk(self, file, data_objects, chunk_index):
             object_data[obj.path] = obj.read_values(file, number_values, self.endianness)
         return RawDataChunk.channel_data(object_data)
 
-    def _read_channel_data_chunk(self, file, data_objects, chunk_index, channel_path):
+    def _read_channel_data_chunk(self, file, data_objects, chunk_index, channel_path, chunk_size):
         """ Read data from a chunk for a single channel
         """
         channel_data = RawChannelDataChunk.empty()
+        current_position = file.tell()
         for obj in data_objects:
             number_values = self._get_channel_number_values(obj, chunk_index)
             if obj.path == channel_path:
+                file.seek(current_position)
                 channel_data = RawChannelDataChunk.channel_data(obj.read_values(file, number_values, self.endianness))
+                current_position = file.tell()
+                break
             elif number_values == obj.number_values:
                 # Seek over data for other channel data
-                file.seek(obj.data_size, os.SEEK_CUR)
-            else:
+                current_position += obj.data_size
+            elif obj.data_type.size is not None:
```

Comment:
Is there a reason you've removed the case where `obj.data_type.size is None`? It looks like there aren't any tests that cover this scenario but I think that is needed to allow skipping over channel data for string channels

Reply:
Ah thanks i didn't think about that. So how to calculate the size of a string channel?
Is it also ``obj.data_size``?
Then it might boil down to
```
            if obj.path == channel_path:
                file.seek(current_position)
                channel_data = RawChannelDataChunk.channel_data(obj.read_values(file, number_values, self.endianness))
                current_position = file.tell()
                break
            elif obj.data_type.size is not None:
                # In last chunk with reduced chunk size
                current_position += obj.data_type.size * number_values
            else:
                current_position += obj.data_size
```

Reply:
After looking into this some more I realised this extra case should never be hit, string channels will usually enter the `elif number_values == obj.number_values` branch, and we don't try to read a truncated segment that contains unsized types due to the check here: https://github.com/adamreeve/npTDMS/blob/88ea6726ca286dcdf9c126038fcb161a6e1b4d70/nptdms/tdms_segment.py#L318-L320

I think we should add an exception just in case this behaviour ever changes in future, maybe something like this:
```python
            if obj.path == channel_path:
                file.seek(current_position)
                channel_data = RawChannelDataChunk.channel_data(obj.read_values(file, number_values, self.endianness))
                break
            elif number_values == obj.number_values:
                # Seek over data for other channel data
                current_position += obj.data_size
            elif obj.data_type.size is not None:
                # In last chunk with reduced chunk size
                current_position += obj.data_type.size * number_values
            else:
                raise Exception("Cannot skip over channel with unsized type in a truncated segment")
```